src/main/java/com/amazon/pvar/tspoc/merlin/solver/flowfunctions/AbstractFlowFunctions.java

```diff
@@ -266,6 +315,71 @@ protected Collection<CallNode> findInvocationsOfFunction(Function function) {
         return solver.getPointsToGraph().getKnownFunctionInvocations(alloc);
     }
 
+    /**
+     * Retrieves the variable associated with the base register for a ReadPropertyNode. The variable is always loaded
+     * into the base register in a ReadVariableNode that occurs one node before the ReadPropertyNode.
```

Comment:
In principle, this could be considered an implementation detail of TAJS, and might change in the future, but if that becomes an issue, we can always search backwards from `n` to find the latest assignment to the base register.

Reply:
For now this seemed like the most lightweight way to do this. If the assumptions I've made about TAJS are ever broken, we can make it more robust like you suggested.